src/raster/r.survey/r.survey.py

```diff
@@ -0,0 +1,1485 @@
+# % key: treesmap
+# % description: Name of the vector layer representing the forested areas
+# % required: no
+# %end
+# %option G_OPT_V_INPUT
+# % key: buildingsmap
+# % description: Name of the vector layer representing the buildings
+# % required: no
+# %end
+# %option
+# % key: obs_heigh
+# % description: observer_elevation
+# % type: double
+# % required: yes
+# % answer: 1.75
+# %end
+# %option
+# % key: treesheigh
+# % description: field of the attribute table containing information about average threes heigh
+# % type: string
+# % required: no
+# %end
+# %option
+# % key: treesheighdeviation
+# % description: field of the attribute table containing information about standard deviation value relate to the heigh
+# % type: string
+# % required: no
+# %end
+# %option
+# % key: buildingsheigh
+# % description: field of the attribute table containing information about buildings heigh
+# % type: string
+# % required: no
+# %end
+# %option
+# % key: obsabselev
+# % description: field of the attribute table containing information about observer absolute elevation above the same datum used by the DEM (e.g. geodetic elevation)
+# % type: string
+# % required: no
+# %end
+# %option
+# % key: layer
+# % description: layer of the attribute table of the point map (can be usefull for obsabselev option)
+# % type: string
+# % answer: 1
+# % required: no
+# %end
+# %option
+# % key: viewangle_threshold
+# % description: cut the output layers at a given threshold value (between 90 and 180 degrees)
+# % gisprompt: 90-180
+# % type: double
+# % required: yes
+# %answer: 90
+# %end
+# %option
+# % key: object_radius
+# % description: radius of the surveyed object in unit of map (default is half the DEM resolution)
+# % type: double
+# % required: no
+# %end
+# %option
+# % key: nprocs
+# % description: Number of processes
+# % answer: 1
+# % type: integer
+# required: yes
+# %end
+# %option
+# % key: memory
+# % description: Amount of memory to use in MB (for r.viewshed analysis)
+# % answer: 500
+# % type: integer
+# required: no
+# %end
+# %flag
+# % key: b
+# % description: Create a simple visible-not visible raster map
+# %end
+# %flag
+# % key: c
+# % description: Consider the curvature of the earth (current ellipsoid)
+# %end
+# %flag
+# % key: d
+# % description: allow only downward view direction (can be used for drone view)
+# %end
+
+
+# Importing modules
+import atexit
+from subprocess import PIPE
+from grass.script import parser, parse_key_val
+from grass.pygrass.modules import Module
+import multiprocessing
+import threading
+import sys
+import grass.script as gscript
+from grass.script import core as grasscore
+from math import pi
+
+# are we in LatLong location?
+s = gscript.read_command("g.proj", flags="j")
+kv = gscript.parse_key_val(s)
+if kv["+proj"] == "longlat":
+    # gscript.fatal(_("This module does not operate in LatLong locations"))
+    gscript.fatal("This module does not operate in LatLong locations")
+    sys.exit()
```

Comment:
```suggestion
```

This needs to go into `main()`, otherwise `g.extension` will fail.
Re-added below.